labelbox/pagination.py

```diff
@@ -64,6 +92,11 @@ def __next__(self):
             if len(page_data) == 0:
                 raise StopIteration()
 
+            if self.cursor_path:
```

Comment:
Can we push all internal cursor state into a single object s.t.

```
class CursorState:
      fetched_all: boolean
      path: something

class OffState:
    _fetched_page: number

PaginationState = Union[CursorState, OffsetState]
```
s.t.  you can't get confused by having the state for both mutually exclusive members floating around

Reply:
Yes, I like this idea.

Reply:
fixed.